pkg/kubelet/cm/dra/manager.go

```diff
@@ -356,7 +354,7 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 			}
 			batches[pluginName] = append(batches[pluginName], claim)
 		}
-		claimInfos[claimInfo.ClaimUID] = claimInfo
+		claimInfos[claimInfo.ClaimUID] = &claimInfo
```

Comment:
Likewise where about not calling `add()` /  `update()` instead.

Reply:
We don't want to remove these claim infos at this point. we'll do it after successful `NodeUnprepareResources` call (see line 389 below). This is symmetrical to the `NodePrepareResources`. We first collect claim infos to the map and then adding them to the cache after successful `NodePrepareResources`.